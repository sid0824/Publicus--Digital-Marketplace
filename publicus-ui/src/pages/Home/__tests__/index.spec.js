import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Home} from "../index";

configure({ adapter: new Adapter() });


describe('Home', () => {

  beforeEach(() => {

  });

  afterEach(() => {

  });

  test('should render authenticated home page', () => {
    const wrapper = shallow(
      <Home
        authenticated
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should render un-authenticated home page', () => {
    const wrapper = shallow(
      <Home
        authenticated={false}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});