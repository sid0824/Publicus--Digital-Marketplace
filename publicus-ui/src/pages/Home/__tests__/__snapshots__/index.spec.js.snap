// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render authenticated home page 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "margin": "auto",
      "paddingBottom": 200,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "marginRight": 50,
      }
    }
  >
    <Icon
      style={
        Object {
          "color": "#1890ff",
          "fontSize": 200,
          "marginTop": 15,
        }
      }
      type="yuque"
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 60,
        }
      }
    >
      Welcome to Publicus!
    </div>
    <div
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      An online tool for creating and publishing forms
      <br />
      Begin by 
      <Link
        to="/createForm"
      >
        creating a form
      </Link>
      <br />
      Or 
      <Link
        to="/marketplace"
      >
        visiting the marketplace
      </Link>
    </div>
  </div>
</div>
`;

exports[`Home should render un-authenticated home page 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "margin": "auto",
      "paddingBottom": 200,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "marginRight": 50,
      }
    }
  >
    <Icon
      style={
        Object {
          "color": "#1890ff",
          "fontSize": 200,
          "marginTop": 15,
        }
      }
      type="yuque"
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 60,
        }
      }
    >
      Welcome to Publicus!
    </div>
    <div
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      An online tool for creating and publishing forms
      <br />
      <Link
        to="/createAccount"
      >
        Create an account
      </Link>
       or 
      <Link
        to="/login"
      >
        login
      </Link>
       to begin
    </div>
  </div>
</div>
`;
